### PR TITLE
fix: delete the daemonset pod on root that is assigned to the leaf node

### LINKED ISSUE #320: daemonset: deleted pod on root is always terminating
**What happened**:
When a pending pod is deleted from root, the pod stays in terminating.

**What you expected to happen**:
it can be really deleted. 

**How to reproduce it (as minimally and precisely as possible)**:
create a daemonset and delete it

**Anything else we need to know?**:

**Environment**:
- Kosmos version:
- Others:

### PR DESCRIPTION
**Pull request by rxy0210**
#### What type of PR is this?
/kind bug

#### What does this PR do?

delete the daemonset pod on root that is assigned to the virtual node

#### Which issue(s) does this PR fix?

Fixes #320 
#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

**Comment by duanmengkk** [comment]
add the issue for the pr

**Comment by rxy0210** [comment]
> add the issue for the pr

i fixed it

File: pkg/clustertree/cluster-manager/controllers/pod/root_pod_controller.go [hunk 1]
```diff
@@ -248,6 +248,10 @@ func (r *RootPodReconciler) SetupWithManager(mgr manager.Manager) error {
 							return true
 						}
 					}
+					_, has := p.Annotations[utils.KosmosNodeOwnedByClusterAnnotations]
+					if (!p.DeletionTimestamp.IsZero()) && (strings.HasPrefix(p.Spec.NodeName, utils.KosmosNodePrefix) || has) {
```
Comment (OrangeBao) [inline pkg/clustertree/cluster-manager/controllers/pod/root_pod_controller.go]: This attribute ```KosmosNodeOwnedByClusterAnnotations```  will only be marked on the node.

File: pkg/clustertree/cluster-manager/controllers/pod/root_pod_controller.go [hunk 2]
```diff
@@ -805,7 +809,7 @@ func (r *RootPodReconciler) DeletePodInLeafCluster(ctx context.Context, lr *leaf
 	leafPod := &corev1.Pod{}
 
 	cleanRootPodFunc := func() error {
-		return DeletePodInRootCluster(ctx, rootnamespacedname, r.Client)
+		return DeletePodInRootCluster(ctx, rootnamespacedname, r.RootClient)
```
Comment (OrangeBao) [inline pkg/clustertree/cluster-manager/controllers/pod/root_pod_controller.go]: ```r.RootClient``` === ```r.Client```

**Pull request merged**